Add Variant feature with request/response handling and tests

marierinden opened the pull request:
### Description

This pull request introduces the Variant feature with the following key components:
- `VariantRequest`, `VariantResponse`, and `VariantResource` classes for managing variant data operations.
- A status transformer for handling response states.
- Comprehensive tests for request/response handling and collections support.

These changes establish the core functionality for variant management and retrieval.

marierinden committed:
Add Variant feature with request/response handling and tests

Introduced VariantRequest, VariantResponse, VariantResource, and related classes to handle variant data operations. Added status transformer, tests for request/response, and support for handling collections. This implements core functionality for managing and retrieving variant details.

fredrik-tveraaen-apiera committed:
Update namespaces in integration test files (#40)

Updated all test files under "tests/Integration" to use the "Tests\Integration" namespace for consistency. This ensures proper test structure alignment and improves organization.

fredrik-tveraaen-apiera committed:
Update VariantRequest to use Product IRI instead of Store IRI

Replaced the store field with product in VariantRequest to align with API requirements. Adjusted related validations, method signatures, and tests accordingly. Added Variant response handling in LdType and integrated variant resource into ApieraSdk.

fredrik-tveraaen-apiera committed:
Add integration tests for Variant resource operations

Added Create, Get, Update, and Delete integration tests for the Variant resource. These tests ensure the correct behavior of variant-related API operations, including proper request handling, response validation, and mocked data assertions.

fredrik-tveraaen-apiera commented on src/DTO/Response/Variant/VariantResponse.php:
```diff
@@ -0,0 +1,171 @@
+final readonly class VariantResponse extends AbstractResponse
+{
+    /**
+     * @param string [] $attributeTerms
```
Remove space

fredrik-tveraaen-apiera commented on src/DTO/Response/Variant/VariantResponse.php:
```diff
@@ -0,0 +1,171 @@
+{
+    /**
+     * @param string [] $attributeTerms
+     * @param string [] $images
```
Remove space

fredrik-tveraaen-apiera commented on src/Enum/VariantStatus.php:
```diff
@@ -0,0 +1,15 @@
+namespace Apiera\Sdk\Enum;
+
+/**
+ * @author Fredrik Tveraaen <fredrik.tveraaen@apiera.io>
```
Wrong signature

fredrik-tveraaen-apiera commented on src/Transformer/VariantStatusTransformer.php:
```diff
@@ -0,0 +1,39 @@
+use Apiera\Sdk\Exception\TransformationException;
+use Apiera\Sdk\Interface\TransformerInterface;
+use Throwable;
+
```
Missing docblock with @author and @since

marierinden committed:
Update author annotations and fix docblock formatting

Updated author tags in `VariantStatusTransformer` and `VariantStatus` to reflect the new contributor. Also corrected docblock formatting in `VariantResponse` for consistency with coding standards.
